source/common/network/connection_balancer_impl.h

```diff
@@ -1,12 +1,34 @@
 #pragma once
 
 #include "envoy/network/connection_balancer.h"
+#include "envoy/registry/registry.h"
+
+#include "source/common/protobuf/protobuf.h"
 
 #include "absl/synchronization/mutex.h"
 
 namespace Envoy {
 namespace Network {
 
+class ConnectionBalanceFactory : public Config::TypedFactory {
```

Comment:
This is on the right track but:
1) The factory should create connection balancers and return a unique_ptr to an interface.
2) The factory should lookup by type_url like all other new factories, which will allow configuration for the balancer. See other examples of factories for how this is done.

/wait